### ISSUE TITLE
0.7.1 OpenApiRequestBody not surfacing Example value in swagger output

### ISSUE THREAD

**Issue by Philcartmell** [Azure/azure-functions-openapi-extension#106]
Package versions

0.7.1-preview
Microsoft.OpenApi 1.2.3
Microsoft.NET.Sdk.Functions 3.0.11

Having added a type that inherits from OpenApiExample<T>

```
public class ChecklistExample : OpenApiExample<Checklist>
{
    public override IOpenApiExample<Checklist> Build(NamingStrategy namingStrategy = null)
    {
        this.Examples.Add(OpenApiExampleResolver.Resolve("sample1","example of checklist",  new Checklist()
        {
            Value = "test"
        }, namingStrategy));

        return this;
    }
}
```

And specified it on the OpenApiRequestBody attribute:

```
[OpenApiRequestBody(contentType: "application/json", bodyType: typeof(Checklist), Example = typeof(ChecklistExample))]
```

The example isn't surfaced in the swagger.json or UI.

If I put a breakpoint inside the Build method() it gets hit and populates the Examples collection.

Struggling to find any more documentation on whether their is additional configuration needed for it to respect the Example value on the attribute.

**Comment by Philcartmell** [comment]
Just realised this is due to the application running in the default version OpenApiVersionType.V2.

To change to V3 override a DefaultOpenApiConfigurationOptions and set the OpenApiVersion property to V3.

[DefaultOpenApiConfigurationOptions ](https://github.com/Azure/azure-functions-openapi-extension/blob/main/samples/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.V3Static/Configurations/OpenApiConfigurationOptions.cs#L11)

**Comment by ferrata** [comment]
I am posting it here for someone who can't use OpenAPI 3 for some reason.

If you need it fixed for OpenAPI 2.0, add the `OpenApiExample` attribute to your data model that has a link to your example class.

```cs
    [OpenApiExample(typeof(ChecklistExample))]
    public class Checklist
    {
        // properties ...
    }
```
